core/src/test/java/uk/gov/gchq/koryphe/impl/function/ParseDateTest.java

```diff
@@ -95,6 +97,47 @@ public void shouldParseDateWithFormat() throws ParseException {
         assertEquals(new SimpleDateFormat("yyyy-MM hh:mm:ss.SSS").parse(input), result);
     }
 
+    @Test
+    public void shouldParseTimestampInMilliseconds() throws ParseException {
+        // Given
+        final ParseDate function = new ParseDate();
+        final String input = "946782245006";
+
+        // When
+        final Date result = function.apply(input);
+
+        // Then
+        assertEquals(new Date(Long.parseLong(input)), result);
+    }
+
+    @Test
+    public void shouldParseTimestampInMicroseconds() throws ParseException {
+        // Given
+        final ParseDate function = new ParseDate();
+        function.setMicroseconds(true);
+        final String input = "946782245006123";
+
+        // When
+        final Date result = function.apply(input);
+
+        // Then
+        assertEquals(Long.parseLong(input), ChronoUnit.MICROS.between(Instant.EPOCH, result.toInstant()));
```

Comment:
This test fails as the Date that comes back can only handle millisecond precision